pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -128,7 +129,11 @@ func createDnsmasqConfiglet(
 	advertizeRouter := true
 	var router string
 
-	if Ipv4Eid {
+	ifindex, _ := devicenetwork.IfnameToIndex(netconf.Port)
```

Comment:
Are you also getting the UI/zedcloud to accept 0.0.0.0 as the gateway/router field?
We should fix this as well.

Here I think it makes more sense to check for netconf.Port == "" (which does assume that the UI/zedcloud doesn't feed the device odd things like "0" or "None" when there is no port). But checking for ifindex can cause odd behaviors if the device is in the processing coming back from pciback so I don't think we should depend on that.

Reply:
it does create error if putting in 0.0.0.0; but if we allow that in UI, i'm still wondering if define the gateway address may be actually useful from dnsmasq point of view, which our attachment to that bridge in dom0 is the 'gateway' address, but it only offers ip address allocation.

Reply:
For the use cases we've seen so far there is a case where a no gateway is useful, which is a Linux domU with two ethernet interfaces where you want the default route only out one of them. For instance, the shopfloor on eth1 is 10.10.10.0/24 and you only want a route for that subnet out eth1.

In that case specifying 0.0.0.0 to mean no default router on that network instance is useful.